odoo/tests/form.py

```diff
@@ -1034,3 +1035,28 @@ def __init__(self, values):
     def __getattr__(self, key):
         val = self.__values[key]
         return Dotter(val) if isinstance(val, dict) else val
+
+
+class Companies:
+    def __init__(cls, env):
+        cls.__env = env
+
+    @property
+    def active_id(self):
+        return self.__env.user.company_id.id
+
+    @property
+    def active_ids(self):
+        return self.__env.user.company_ids.ids
+
+    @property
+    def allowed_ids(self):
+        return self.__env.companies.ids
+
+    @property
+    def multi_company(self):
+        return len(self.__env.companies) > 1
+
+    def has(self, ids, fieldName, value):
+        ids = [ids] if isinstance(ids, int) else ids
+        return any(self.__env.companies, lambda c: c.id in ids and getattr(c, fieldName, False) == value)
```

Comment:
```suggestion
    def has(self, ids, field_name, value):
        if field_name not in companies._fields:
            return False
        ids = [ids] if isinstance(ids, int) else ids
        return any(company[field_name] == value for company in self.__env.user.company_ids if company.id in ids)
```

Reply:
I have a doubt about `self.__env.user.company_ids`. Because if a record is linked to another company, to which the user cannot connect (but the user can have access to this record), the information returned must be correct.
What do you think?

`companies.has(company_id, 'xxxx', True)`

Reply:
The `has` function in JS was developed as :
```js
has: (ids, field, value) => {
            ids = typeof ids === "number" ? [ids] : ids || [];
            return allowedCompanies.some((c) => ids.includes(c.id) && c[field] === value);
        }
```

We use the `allowedCompanies` (the python `self.__env.user.company_ids`).

> Because if a record is linked to another company, to which the user cannot connect (but the user can have access to this record), the information returned must be correct.

Unfortunately, on JS only the company information that came in the session_info is available, which is the allowed companies of the user. In the case you describe, the information will not be available.